rotkehlchen/inquirer.py

```diff
@@ -558,18 +559,31 @@ def find_curve_pool_price(
         """
         assert self._ethereum is not None, 'Inquirer ethereum manager should have been initialized'  # noqa: E501
 
-        pools = get_curve_pools()
-        if lp_token.evm_address not in pools:
+        pool_addr_in_cache = GlobalDBHandler().get_general_cache_value(
+            cache_type=GeneralCacheType.CURVE_POOL_ADDR,
+            name=lp_token.evm_address,
+        )
+        if pool_addr_in_cache is None:
```

Comment:
Why stop if the pool address is not cached? Can't you ask for it here?

Reply:
In general yes, it is possible to ask. But I think it will be a bit overengineering since the probability of this is quite low. Or is it still better to ask even with a low chance of this happening?

Reply:
Overengineering is when you make something more complicated than it should be.

Here you are not making something complicated. The way it's written you are making a price query fail completely if the cache is not there. Right? And cache is queries only once every 2 weeks.

So when this happens the user has no chance to get any curve prices for 2 weeks?

Reply:
Yes.

But I mean if low probability of this doesn't matter, I will surely add an extra call here

Reply:
But question... If we are going to be updating cache here, do we need the periodic task? Won't it be unneeded duplication?

Reply:
Why duplication? The cache update is (I hope) its own independent function/logic.

It can be ran from various places. A periodic task runs periodically. This is user-initiated. Two different things, no?

Reply:
The thing that I meant is that if we will query all unknown pools at the price query moment periodic task won't be really needed since we will be able to get the all the pools without it.

But then I thought that it is still going to save us time, so perhaps makes sense to have it.

Reply:
Oh shit, I just found out that I lost significant time yesterday. I was trying to answer the question 'Why stop if the pool address is not cached? Can't you ask for it here?" and was thinking about whether we should ask for the pool address here or not and then spent quite some time on implementing it.

But.

I forgot about the fact that `pool_addr_in_cache` should never be None here since we cache it at the same time when we add the `lp token` to our database and mark this `lp token` as curve's. So basically it means that if there is a token marked as curve's there is definitely all the related info in the cache.